### DISCUSSION TITLE
It's possible to sync 3 databases in unidirectional way?

### DISCUSSION THREAD

**Discussion by igorgomeslima** [Mimetis/Dotmim.Sync#1037] [General]
We have 3 SQLS databases and we need to sync when data changes on any of these databases.

**Like:**

> Data changed at `Database1` = Sync to `Database2` & `Database3`.
> Data changed at `Database2` = Sync to `Database1` & `Database3`.
> Data changed at `Database3` = Sync to `Database1` & `Database2`.

I've already tried:

```
var database1 = new SqlSyncProvider(@"Server=localhost, 1433;Database=DataSync;User Id=sa;Password=👀;TrustServerCertificate=True;");
var database2 = new SqlSyncProvider(@"Server=localhost, 1434;Database=DataSync;User Id=sa;Password=👀;TrustServerCertificate=True;");
var database3 = new SqlSyncProvider(@"Server=localhost, 1435;Database=DataSync;User Id=sa;Password=👀;TrustServerCertificate=True;");

// Tables involved in the sync process:
var setup = new SyncSetup("balance");

// Sync agents
var agent1 = new SyncAgent(database2, database1);
var agent2 = new SyncAgent(database3, database2);

do
{
    var result1 = await agent1.SynchronizeAsync(setup);
    Console.WriteLine(result1);
    var result2 = await agent2.SynchronizeAsync(setup);
    Console.WriteLine(result2);

} while (Console.ReadKey().Key != ConsoleKey.Escape);
```

Why when data changes at `database1` the data sync only occurs on `database2` and not at `database3`? Since we have an `SyncAgent` to sync `database3 & database2 `(default mode `Bidirectional`). Seem like `agent2` do not sync data, like a "cyclic" data change.

What's is the correct way to achieve the sync goal, it's possible?

Cheers!

**Comment by Mimetis** [comment]
Peer to peer sync is not working With DMS.
You need to have one single DB as server, and all the others DB as clients, syncing only with the server. This is called Hub & Spoke topology

**Reply by igorgomeslima** [reply]
Thanks @Mimetis!
